app/services/catalog/add_to_order.rb

```diff
@@ -8,6 +8,8 @@ def initialize(params)
 
     def process
       order = Order.find_by!(:id => @params[:order_id])
+      # override the service_plan_ref (even if they pass one in)
+      @params[:service_plan_ref] = service_plan_ref
```

Comment:
Should we just delete `service_plan_ref` if it comes in?  Possibly also remove it from the `order_items_params` method as permitted? ( Also pull it from the openapi spec? )

Reply:
Agree with @syncrou here, I don't think there's a reason we need it in the `order_items_params` or the openapi spec anymore, right?

Reply:
Please verify that removing `service_plan_ref` from the spec will not break current UI/API functionality.  We do not want to break backwards compatibility on this one.  That would be a followup PR with associated UI PR.

Reply:
Yes - it will break the UI if I remove it from the spec. I'm going to leave it for now and just handle it on the backend by deleting the param and removing it from the `permit!` list. 

It will then be a one-line removal in the service and remove it from the spec definition once we're ready.

Reply:
You can create a WIP PR for that one-line change and cc @Hyperkid123 on it so we can get the UI done.  No need to wait to create the PR, just want to hold off merging until we are in sync.

Reply:
PR #523 opened for that change.